File: fux.js
const axios = require('axios');
const cheerio = require('cheerio');

module.exports = async function fux(query) {
  const url = `https://www.fux.com/videos/search/${encodeURIComponent(query)}`;
  const results = [];

  try {
    const { data } = await axios.get(url);
    const $ = cheerio.load(data);

    $('.item-video').each((i, el) => {
      const title = $(el).find('.title').text().trim();
      const href = $(el).find('a').attr('href');
      const duration = $(el).find('.duration').text().trim();

      if (title && href) {
        results.push({
          title,
          url: 'https://www.fux.com' + href,
          duration,
          source: "Fux"
        });
      }
    });
  } catch (err) {
    console.error("fux error:", err.message);
  }

  return results;
};

File: scrapers/fux.py
import requests
from bs4 import BeautifulSoup

def scrape_fux(query, mode="straight", page=1):
    results = []
    if mode == "gay":
        query = f"gay {query}"
    elif mode == "trans":
        query = f"trans {query}"
    url = f"https://www.fux.com/search/videos/{query}/?page={page}"
    headers = {'User-Agent': 'Mozilla/5.0'}
    r = requests.get(url, headers=headers, timeout=10)
    soup = BeautifulSoup(r.content, "html.parser")
    for vid in soup.select("div.item"):
        try:
            a = vid.select_one("a")
            if not a or not a.has_attr("href"):
                continue
            title = a.get('title') or a.text.strip()
            video_url = "https://www.fux.com" + a['href']
            img = vid.select_one("img")
            preview = img.get("data-src") or img.get("src")
            results.append({
                "title": title,
                "url": video_url,
                "preview": preview,
                "source": f"Fux ({mode})"
            })
        except Exception:
            continue
    return results
